cellprofiler/pipeline.py

```diff
@@ -1117,6 +1117,87 @@ def encode_txt(self, s):
         s = s.replace('[', '\\x5B').replace(']', '\\x5D')
         return s
 
+    def to_dictionary(self):
+        pipeline_dictionary = {
+            "cookie": COOKIE,
+            "modules": [],
+            "version": cellprofiler.__version__
+        }
+
+        for module in self.__modules:
+            module_dictionary = {
+                "batch_state": str(getattr(module, "batch_state")),
+                "displayed": getattr(module, "show_window"),
+                "enabled": getattr(module, "enabled"),
+                "name": module.module_name,
+                "notes": getattr(module, "notes"),
+                "paused": getattr(module, "wants_pause"),
+                "settings": {},
+                "version": getattr(module, "variable_revision_number")
+            }
+
+            for setting in module.settings():
+                # FIXME: Why does CellProfiler have hidden settings?
+                if not setting.text:
+                    continue
+
+                v = self.encode_txt(setting.unicode_value.encode('utf-8'))
```

Comment:
The thing about encoding the values as utf-8 means that the actual pipeline file isn't really human readable. I was thinking - wouldn't it be possible to save the file as utf-8 and just keep these as unicode values?